Source/cmake/WebKitCommon.cmake

```diff
@@ -258,6 +261,67 @@ if (NOT HAS_RUN_WEBKIT_COMMON)
                     "${CMAKE_SOURCE_DIR}"
                     "${CMAKE_BINARY_DIR}"
         )
+    elseif (CMAKE_EXPORT_COMPILE_COMMANDS)
+        # Create a simple symlink in DeveloperTools/compile_commands.json so that
+        # it can be relied upon regardless of unified or non-unified builds.
+        add_custom_command(
+            OUTPUT ${CMAKE_BINARY_DIR}/DeveloperTools/compile_commands.json
+            DEPENDS ${CMAKE_BINARY_DIR}/compile_commands.json
+            COMMAND ${CMAKE_COMMAND} -E create_symlink
+                    ../compile_commands.json
+                    ${CMAKE_BINARY_DIR}/DeveloperTools/compile_commands.json
+            VERBATIM
+        )
+        add_custom_target(SymlinkCompileCommandsInDeveloperTools
+            ALL
+            DEPENDS ${CMAKE_BINARY_DIR}/DeveloperTools/compile_commands.json
+        )
+    endif ()
+
+    set(webkitbuild_dir ${CMAKE_SOURCE_DIR}/WebKitBuild)
+    cmake_path(IS_PREFIX webkitbuild_dir ${CMAKE_BINARY_DIR} NORMALIZE clangd_auto_setup_default)
+    set(CLANGD_AUTO_SETUP ${clangd_auto_setup_default} CACHE BOOL
+        "Install a .clangd configuration file and a compile_commands.json symlink "
+        "in the root of the source tree to have out-of-the-box code completion "
+        "in editors.")
+
+    if (CLANGD_AUTO_SETUP)
+        # compile_commands.conf
+        add_custom_command(
+            OUTPUT ${CMAKE_SOURCE_DIR}/compile_commands.conf
+            DEPENDS ${TOOLS_DIR}/clangd/update-compile-commands-conf
+                    ${TOOLS_DIR}/clangd/compile_commands.conf.example
+            COMMAND ${Python_EXECUTABLE}
+                    ${TOOLS_DIR}/clangd/update-compile-commands-conf
+                    ${CMAKE_SOURCE_DIR}/compile_commands.conf
```

Comment:
Do you want to pass the path of compile_commands.conf.example as an argument?
https://github.com/WebKit/WebKit/pull/29313#discussion_r1624584591

Reply:
That would make sense indeed.

Reply:
(Update) This is done in the new version.